dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -195,6 +200,25 @@ namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
         return *this;
     }
 
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddTextBox(hstring id)
+    {
+        ThrowIfMaxInputItemsExceeded();
+        THROW_HR_IF_MSG(E_INVALIDARG, id.empty(), "You must provide an id for the TextBox");
+
+
+        m_textBoxList.push_back(wil::str_printf<std::wstring>(L"<input id='%ls' type='text'/>", id.c_str()));
```

Comment:
Is ID restricted to never allow standard entities that need to be escaped?

standard entities are the gang of 5 -- greater-than, less-than, apostraphe, quote, ampersand

Reply:
Looks like it doesn't like " & <, but also it doesn't mind empty string. I am introducing an escape character map in my PR :https://github.com/microsoft/WindowsAppSDK/pull/2805 @loneursid. We can add these to this map when my PR merges.

Reply:
We are using percent encoding for our special characters: https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding

Go ahead and merge this PR when ready, I'll raise another PR regarding this issue since it would require more of a sweeping change in AppNotificationBuilder.

Reply:
Yeah, I'll do that. I'll merge this as is, since it's approved and the build is passing and raise a new PR to address nits from this and other PRs in the series

Reply:
List of xml escape chars:
"   &quot;
'   &apos;
<   &lt;
>   &gt;
&   &amp;

Reply:
Addressing in: https://github.com/microsoft/WindowsAppSDK/pull/2813

Reply:
Thanks @DrusTheAxe for catching this. It totally slipped past me.

Since we put anything that comes to us through the builder API into single quotes, technically, that the only character we need to escape, right? Or am I missing something?